### ISSUE TITLE
Cassandra StatefulSet: Pod errors: Unschedulable

### ISSUE THREAD

**Issue by didip** [kubernetes/examples#225]
Tested on GCP Kubernetes:
```
Master version                         1.9.6-gke.0
Endpoint	                                 35.184.206.250
Client certificate	                 Enabled
Kubernetes alpha features	Disabled
Total size	                                 3
Master zone	                         us-central1-a
Node zones	                         us-central1-a
Network	                                 default
Subnet	                                 default
Alias IP ranges	                Disabled
Container address range	10.12.0.0/14
Stackdriver Logging	        Enabled
Stackdriver Monitoring	        Enabled
Private cluster                         Disabled
Master authorized networks	Disabled
Network policy	                Disabled
Legacy authorization	        Disabled
Maintenance window	        Any time
```

I followed the following tutorial:

* https://kubernetes.io/docs/tutorials/stateful-application/cassandra/

* https://github.com/kubernetes/examples/tree/master/cassandra

The `cassandra-data-cassandra-0` volume continuously spins in pending phase.

How can I fix it? I don't see other events to clue me in.

**Comment by yanniszark** [comment]
The aforementioned tutorial is made for Minikube, but it seems you are using Kubernetes on GCP. 
The main problem is the use of the hostpath provisioner, as hostpath is not meant to be used in multi-node clusters and can give you all sorts of headaches (see [here at 8:35](https://www.youtube.com/watch?v=eqkgiPppZN8&t=2141s) ). 
You have 2 options to continue: 
1) Use a GCE provisioner to provide cassandra with storage. In this case you need to change the StatefulSet yaml to point to the correct PV.
2) Use [Local Storage](https://kubernetes.io/blog/2018/04/13/local-persistent-volumes-beta/) (Beta in 1.10). Cassandra is considered a good use-case for this, as it replicates its data, so it can handle node failures. Currently there is a static provisioner for local storage available from the external-storage repo. I have a working demo of this and I would be glad to make a guide if there is interest. 

In conclusion, I would propose you use the GCE provisioner to get started, as it is most likely the quicker and easier way to get it working. Then, if you want, you can try it with Local Storage.

**Comment by ahmetb** [comment]
This will be tricky. This repo provides examples neutral to any cloud provider or vendor-specific steps.
